### ISSUE TITLE
Deleting and re-drafting a reply breaks thread

### ISSUE THREAD

**Issue by durka** [elk-zone/elk#1280]
### Pre-Checks

- [X] Check that there isn't [already an issue](https://github.com/elk-zone/elk/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/elk-zone/elk/discussions) or join our [Discord Chat Server](https://chat.elk.zone).
- [X] Providing a screenshot or video to reproduce the issue or show visually what was meant.
- [X] I am willing to provide a PR.

### Describe the bug

STR:

1. Reply to a toot.
2. Select "delete and re-draft" on the reply.
3. Change something, or don't.
4. Publish the re-drafted reply

Expected result:
5. Re-drafted toot is tooted as a reply to the original toot.

Actual result:
5. Re-drafted toot is tooted as a standalone toot.
6. @-reply at the beginning of the re-drafted toot is broken -- the first @ is dropped.

### Reproduction video or screenshot

I can make a video if necessary but I think the description above is clear -- let me know if not?

### Additional Context

If I can figure out how, I'd be glad to contribute to fixing this.

**Comment by patak-dev** [comment]
Would you test using https://main.elk.zone instead of https://elk.zone?

**Comment by durka** [comment]
On main.elk.zone it seems to still break the thread although the leading
"@" did not get deleted. By the way, for both of these tests I used the
mobile web browser on Android. I will test more thoroughly this weekend.

**Comment by jonasnordlund** [comment]
Yep, confirmed this too. I first thought it might be a display like #399 (which I honestly just got on the main elk.zone site despite that one marked as closed). But refreshing showed it as becoming a separate post, and it was also not underneath the original post.

**Comment by durka** [comment]
I tested again using elk.zone and main.elk.zone. In both cases, invoking "delete and re-draft" on a reply results in the re-published toot *not* being attached to its original parent.

**Comment by durka** [comment]
This was later duplicated by #2023 and fixed in #2028, thanks @alexzhang1030
